jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/ui/UiUtilsTest.kt

```diff
@@ -36,4 +43,81 @@ class UiUtilsTest {
         assertThat(listModel.find { it == "hello" }).isEqualTo("hello")
         assertThat(listModel.find { it == "bye" }).isEqualTo(null)
     }
+
+    @Test
+    fun comboBoxPopulation_useDefaultSelected() {
+        val comboBox = ComboBox<String>()
+        val items = listOf("foo", "bar", "baz")
+
+        comboBox.model.selectedItem = "foo"
+        comboBox.populateValues(defaultSelected = "bar") { items }
+
+        waitForPopulationComplete(comboBox, items.size)
+        assertThat(comboBox.selectedItem).isEqualTo("bar")
+    }
+
+    @Test
+    fun comboBoxPopulation_overrideDefaultSelected() {
+        val comboBox = ComboBox<String>()
+        val items = listOf("foo", "bar", "baz")
+
+        comboBox.model.selectedItem = "foo"
+        comboBox.populateValues(defaultSelected = "bar", overrideDefaultSelected = true) { items }
+
+        waitForPopulationComplete(comboBox, items.size)
+        assertThat(comboBox.selectedItem).isEqualTo("foo")
+    }
+
+    @Test
+    fun comboBoxPopulation_useDefaultSelectedWhenPreviouslySelectedIsNull() {
+        val comboBox = ComboBox<String>()
+        val items = listOf("foo", "bar", "baz")
+
+        comboBox.populateValues(defaultSelected = "bar", overrideDefaultSelected = true) { items }
+
+        waitForPopulationComplete(comboBox, items.size)
+        assertThat(comboBox.selectedItem).isEqualTo("bar")
+    }
+
+    @Test
+    fun comboBoxPopulation_notUpdateState() {
+        val comboBox = ComboBox<String>()
+        val items = listOf("foo", "bar", "baz")
+
+        comboBox.isEnabled = false
+        comboBox.populateValues(updateStatus = false) { items }
+
+        waitForPopulationComplete(comboBox, items.size)
+        assertThat(comboBox.isEnabled).isEqualTo(false)
+    }
+
+    @Test
+    fun comboBoxPopulation_updateStateToTrueWhenItemsAreNotEmpty() {
+        val comboBox = ComboBox<String>()
+        val items = listOf("foo", "bar", "baz")
+
+        comboBox.isEnabled = false
+        comboBox.populateValues(updateStatus = true) { items }
+
+        waitForPopulationComplete(comboBox, items.size)
+        assertThat(comboBox.isEnabled).isEqualTo(true)
+    }
+
+    @Test
+    fun comboBoxPopulation_updateStateToFalseWhenItemsAreEmpty() {
+        val comboBox = ComboBox<String>(arrayOf("foo", "bar"))
+
+        comboBox.isEnabled = true
+        comboBox.populateValues(updateStatus = true) { listOf() }
+
+        waitForPopulationComplete(comboBox, 0)
+        assertThat(comboBox.isEnabled).isEqualTo(false)
+    }
+
+    // Wait for the combo box population complete by detecting the item count
+    private fun <T> waitForPopulationComplete(comboBox: ComboBox<T>, count: Int) {
+        while (comboBox.itemCount != count) {
+            Thread.sleep(1000)
```

Comment:
let's make this much smaller - 50-100ms maybe

Reply:
Updated to 100ms